### PR TITLE
TECH-1866: Add Quickstart Guide to Flow gem Readme

### PR DESCRIPTION
**Pull request by tbconroy**
My pass at a "quickstart example" for the README.

### CODE REVIEW

**Comment by garside** [comment]
@tbconroy I could kinda but the separate file thing, but I dunno. It feels kinda nice to have the jump start guide be on the main page. might help contextualize things a bit more?

I think maybe the most ideal form would be like a `quickstart.md` and use like markdown partial rendering if that's a thing to include it in the readme on the githubz? I dunno if that's supported, I've never done it.

File: README.md [hunk 1]
```diff
@@ -84,6 +85,166 @@ Flow is a [SOLID](https://en.wikipedia.org/wiki/SOLID) implementation of the [Co
...
+  # state accessor methods
+  state_reader :order
+  state_reader :user
+  state_reader :payment_method
+
+  state_writer :charge
+
+  def behavior
+    # payment_method is an optional state input, so it may not be present
+    payment_method_to_charge = payment_method.present? payment_method : user.default_payment_method
```
[the remark this answers is not in the document]
Comment (tbconroy) [reply README.md]: Pulling that out into a private method makes sense. The reason I initially used the not ideal name `payment_method_to_charge` was to differentiate it from the state accessor  `payment_method`. I was afraid a flow newbie be confused with how state readers work if the example overloaded that name.

File: README.md [hunk 2]
```diff
@@ -84,6 +85,166 @@ Flow is a [SOLID](https://en.wikipedia.org/wiki/SOLID) implementation of the [Co
...
+```bash
+$ rails generate flow:state Charge
+```
+
+```ruby
+# app/states/charge_state.rb
+
+class ChargeState < ApplicationState
+  # required state input, will raise error if missing
+  argument :order
```
[the remark this answers is not in the document]
Comment (tbconroy) [reply README.md]: I actually have never used Yarddoc 😬, do you have an example of what you are looking for here 😃

File: README.md [hunk 3]
```diff
@@ -84,6 +85,166 @@ Flow is a [SOLID](https://en.wikipedia.org/wiki/SOLID) implementation of the [Co
...
+  option :payment_method
+
+  # expected output from the flow
+  output :charge
+end
+```
+
+### Operations
+
+Define operation that can _operates_ on the state using the `behavior` method. Accessor methods allow operations to read and write to state:
```
[the remark this answers is not in the document]
Comment (tbconroy) [reply README.md]: Made some changes to this copy, let me know what you think!

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tbconroy) [reply README.md]: I took a stab at what I think you wanted: https://rubydoc.info/gems/yard/file/docs/GettingStarted.md#Documenting_Attributes
under `Documenting Custom DSL Methods`

**Comment by tbconroy** [comment]
@garside pinging you on this, I _think_ I addressed all the changes you wanted. The only thing I wasn't 💯 percent on was what you wanted re example yard documentation.

**Pull request merged**